dagian/data_definition.py

```diff
@@ -125,8 +125,7 @@ def eval_data_definition(self, args):
             elif isinstance(arg, basestring):
                 new_args[key] = arg.format(**args)
             else:
-                raise ValueError(
-                    "The values in RequirementDefinition.args can only be Argument or str.")
+                new_args[key] = arg
```

Comment:
We should check whether `arg` is hashable, so you should put `new_arg[key] = arg` in `elif isinstance(arg, collections.Hashable):`, and change the error message in `else:` to `The values in RequirementDefinition.args can only be Argument or hashable.`